app/models/ems_refresh/parsers/kubernetes.rb

```diff
@@ -121,9 +121,13 @@ def parse_node(node)
         parse_node_condition(condition)
       end
 
-      # Searching for the underlying instance. At the moment this search strategy
-      # is tested only with OpenStack.
-      vms = VmOpenstack.where(:uid_ems => new_result[:identity_system].downcase)
+      # Establish a relationship between this node and the vm its on (if its in the system)
+      # supported relationships: oVirt and Openstack
+      types = [ManageIQ::Providers::Redhat::InfraManager::Vm.name,
+               VmOpenstack.name]
+
+      # Searching for the underlying instance for Openstack or oVirt.
+      vms = Vm.where(:uid_ems => new_result[:identity_system].downcase, :type => types)
```

Comment:
> My guess is that there probably is some additional overhead head here.

Certainly not enough to worry about.

@Fryguy we'd need to work out the intersection of the `where` condition vs the inheritance hierarchy condition... including in cases where the condition was e.g., `where(:type => 'A'..'M')`. Punting that work to PG seems the safest bet... and the fact that *it* is choosing to just evaluate the double-condition (which, let's be honest, will be Very Quick Indeed) appears to support the idea that it's not worth bothering to try to be cleverer.

If the second condition were `where(:type => 'ASingleString')`, then I'd expect the PostgreSQL planner to eliminate the longer conditional check through equivalence classes. Not to mention that any formulation that happens to permit use of an index will still manage to do so, even if a secondary useless (strictly more generous) conditional filter is applied to the matching rows.